Feature/author post

Paras-Wednesday opened the pull request:
### Ticket Link
---------------------------------------------------


### Related Links
---------------------------------------------------


### Description
- Adds the migration and seed for the new models `posts` and `authors`
- Fixes documentation for setup and script for setup


### Steps to Reproduce / Test
---------------------------------------------------

### Request
---------------------------------------------------


### Response
---------------------------------------------------

Paras-Wednesday committed:
docs: fix Formatting, fix Seed directives

Paras-Wednesday committed:
fix: precommit script, docker port expose

- The precommit script should include installation for `go-commitlinter`
- The postgres container exposes the port for testing locally

Paras-Wednesday committed:
feat(models): generate models, update dependency

- Update the `strmangle` dependency from 0.0.4 to 0.0.6 as older version
was generating errorneous code
- Update the `tools.go` to import `v4` for `sqlboiler`
- Generate models and write migration script for authors and posts

Paras-Wednesday committed:
refactor: imporved error message for debugging

- Add few verbose error message for better debugging

Paras-Wednesday committed:
refactor: add db connection before seeding users

- user's role needs to be fetched from db, so add db conn for the same

Paras-Wednesday committed:
feat: seeders for `authors` and `posts`

rajatdangat-ws commented on cmd/seeder/exec/seed.go:
```diff
@@ -36,8 +36,9 @@ func main() {
 		err := cmd.Run()
 		if err != nil {
-			log.Fatal(err)
+			fmt.Printf("file: %s, out: %s, err: %s\n", file.Name(), outb.String(), errb.String())
```
Please remove log if not required.

rajatdangat-ws commented on cmd/gendbmdls/main.go:
```diff
@@ -72,14 +74,17 @@ func main() {
 		Imports:         configureImports(),
 	})
 	if err != nil {
+		fmt.Println("error in setting up boilingcore command state")
```
Do we need these println's?

Paras-Wednesday replied:
Although the `Println` itself is not required, but it should still be helpful in adding context to panic

rajatdangat-ws replied:
Yes makes sense. We can keep it.

rajatdangat-ws commented on cmd/seeder/v2/2_users.go:
```diff
@@ -14,11 +15,10 @@ import (
-	// }
+	err := config.LoadEnv()
+	if err != nil {
+		log.Println(err)
```
Maybe we should return and let user know with an error if the env variables are not loaded correctly since next we are trying to connect with the DB using the env variables.

Paras-Wednesday replied:
Yeah, here it should `Fatal` out, as this is `main` func and without `db` we wouldn't be able to fetch the `users` anyway

rajatdangat-ws commented on cmd/seeder/v4/4_posts.go:
```diff
@@ -0,0 +1,31 @@
+func main() {
+	err := config.LoadEnv()
+	if err != nil {
+		log.Println(err)
```
Maybe do a log.Fatalf here since db connection will require the env variables.

Paras-Wednesday committed:
refactor: imporeve error handling and error messages

rajatdangat-ws commented on cmd/seeder/v2/2_users.go:
```diff
@@ -17,7 +17,7 @@ import (
 	err := config.LoadEnv()
 	if err != nil {
-		log.Println(err)
+		log.Fatalf("couldn't load the evn: %s", err)
```
Please fix the typo in the message

rajatdangat-ws commented on cmd/seeder/v4/4_posts.go:
```diff
@@ -15,7 +15,7 @@ import (
 	err := config.LoadEnv()
 	if err != nil {
-		log.Println(err)
+		log.Fatalf("couldn't load the evn: %s", err)
```
Please fix the typo in the message

Paras-Wednesday committed:
fix: typo for the `env`

Paras-Wednesday committed:
fix: db creds injection handling in `LoadEnv`

- Now `LoadEnv` will verify if db cred injection is required or not
- Add helper function to extract the db creds from secret manager
